### PR TITLE
BUGFIX/MEDIUM(SDS): Use Healthcheck for run and tests

### PR DESCRIPTION
**Pull request by dduportal**
##### SUMMARY

- As an OpenIO administrator
  When I execute the OpenIO Docker image for evaluation purpose
  And the services are ready to be used in the started container,
  Then I expect it to be healthy from Docker Daemon point of view

- The healthcheck is a regular HTTP check on the S3 gateway. Test harness updated

- Updated READMEs (forgotten on previous commits)

 ##### IMPACT

- Startup of the image is now a little slower, time for the healthcheck to be ok

 ##### ADDITIONAL INFORMATION

- Documentation of the quick start should be checked with this new image

Signed-off-by: dduportal <damien.duportal@openio.io>

### CODE REVIEW

File: openio-sds/18.10/sds-healthcheck.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -eux -o pipefail
+
+CONTAINER_IP="$(hostname -i)"
+
+command -v curl >/dev/null 2>/dev/null || (echo "ERROR: No curl command found." && exit 1)
```
Comment (vincent-legoll) [inline openio-sds/18.10/sds-healthcheck.sh:7]: groupe tes commandes avec {} plutot que () pour éviter de fork+exec un sous shell

et tu peux remplacer ton 2>/dev/null par 2>&1 (ça passe shellcheck)

File: openio-sds/18.10/test.sh [hunk 2]
```diff
@@ -7,7 +7,14 @@ CURRENT_DIR="$(cd "$(dirname "$0")" && pwd -P)"
...
 test -n "${DOCKER_IMAGE_NAME}" || ( echo "Error: variable DOCKER_IMAGE_NAME not set. Exiting." && exit 1 )
 
+clean_sut() {
+    docker kill "${1}"
+    docker rm -f "${1}"
+    docker system prune -f --volumes
+}
+
 # Launch an openio-sds container to test
+clean_sut "${DOCKER_TEST_CONTAINER_NAME}" || true # Never fail to cleanup
```
Comment (vincent-legoll) [inline openio-sds/18.10/test.sh:17]: s/ to//

File: openio-sds/18.10/test.sh [hunk 3]
```diff
@@ -20,4 +27,12 @@ docker run --rm -t \
     -e "SUT_ID=${DOCKER_TEST_CONTAINER_NAME}" \
     -e SUT_IP="$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' "${DOCKER_TEST_CONTAINER_NAME}")" \
     -v /var/run/docker.sock:/var/run/docker.sock \
-    "${TESTER_IMAGE_NAME}"
+    "${TESTER_IMAGE_NAME}" \
+|| (
```
Comment (vincent-legoll) [inline openio-sds/18.10/test.sh:31]: () => {}

File: openio-sds/18.10/test.sh [hunk 4]
```diff
@@ -20,4 +27,12 @@ docker run --rm -t \
...
+    echo "== Tests failure: please check SUT logs below:"
+    docker logs "${DOCKER_TEST_CONTAINER_NAME}"
+    echo "=============================================="
+    clean_sut "${DOCKER_TEST_CONTAINER_NAME}" || true # Never fail to cleanup
+    exit 1
+)
+clean_sut "${DOCKER_TEST_CONTAINER_NAME}" || true # Never fail to cleanup
```
Comment (vincent-legoll) [inline openio-sds/18.10/test.sh:38]: idem

File: openio-sds/19.04/sds-healthcheck.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
Comment (vincent-legoll) [inline openio-sds/19.04/sds-healthcheck.sh:1]: C'est normal cette duplication (c'est le meme script, non ?) ?

File: openio-sds/18.10/sds-healthcheck.sh (hunk 1 above)
Comment (dduportal) [reply openio-sds/18.10/sds-healthcheck.sh:7]: Self reference pour plus tard:
- https://www.tldp.org/LDP/abs/html/special-chars.html#CODEBLOCKREF (`{}` for bash)
- https://www.tldp.org/LDP/abs/html/special-chars.html#PARENSREF (`()` for bash)

File: openio-sds/18.10/test.sh (hunk 2 above)
Comment (dduportal) [reply openio-sds/18.10/test.sh:17]: Sur la ligne de commentaire ?

Comment (vincent-legoll) [inline openio-sds/18.10/test.sh:17]:
```suggestion
clean_sut "${DOCKER_TEST_CONTAINER_NAME}" || true # Never fail cleanup
```

**Comment by dduportal** [comment]
- commit message updated (but I feel like it is a waste of time as we end up squashing it and the PR message becomes the resulting commit message)
- As @cdelgehier pointed out a few weeks ago, I forgot to update the READMEs when updating the build process: it is now fixed.

**Pull request merged**